HHS/OPRE-OPS
# Adding /cans/ endpoints and service layer refactor

## What changed

I added POST, PUT, PATCH, and DELETE /cans/ as well as refactoring the api endpoints to go through a service layer and isolate business logic. All unit tests were updated to support this. The obligate_by property was also added to the CAN schema at Frank's request to support the consumption of the GET /cans/ endpoint in our related FE work. Finally openapi doc will be updated monday morning to include the new endpoints. 
## Issue

OPS-2781
## How to test

1. Run unit tests to ensure all new backend and existing functionality is still working
2. Run cypress tests to verify that the front end still is interacting with existing GET /cans/ endpoints correctly. 

## Definition of Done Checklist
